### PR TITLE
Add Build Enviroment Vars to Cloud Functions 

### PR DESCRIPTION
**Pull request by upodroid**
Fixes: https://github.com/hashicorp/terraform-provider-google/issues/7589

Also, I bumped nodejs to 10 as 8 can't be created any more.

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
functions: added `build_environment_variables` field to `google_cloudfunction_function`
```

### CODE REVIEW

File: third_party/terraform/tests/resource_cloudfunctions_function_test.go.erb [hunk 1]
```diff
@@ -160,9 +160,10 @@ func TestAccCloudFunctionsFunction_basic(t *testing.T) {
 				),
 			},
 			{
-				ResourceName:      funcResourceName,
-				ImportState:       true,
-				ImportStateVerify: true,
+				ResourceName:            funcResourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"build_environment_variables"},
```
Comment (c2thorn) [inline third_party/terraform/tests/resource_cloudfunctions_function_test.go.erb:166]: @upodroid It isn't apparent to me why the field wouldn't match during an import. Mind filling me in?

Comment (upodroid) [reply third_party/terraform/tests/resource_cloudfunctions_function_test.go.erb:166]: The API doesn't return it.

```
2020/10/21 11:25:23 [DEBUG] Google API Request Details:
---[ REQUEST ]---------------------------------------
GET /v1/projects/REDACTED/locations/us-central1/functions/tf-test-6onvdmxcmr?alt=json&prettyPrint=false HTTP/1.1
Host: cloudfunctions.googleapis.com
User-Agent: google-api-go-client/0.5 Terraform/0.12.29 (+https://www.terraform.io) Terraform-Plugin-SDK/2.0.3 terraform-provider-google-beta/acc 
X-Goog-Api-Client: gl-go/1.14.4 gdcl/20201009
Accept-Encoding: gzip


-----------------------------------------------------
2020/10/21 11:25:23 [DEBUG] Google API Response Details:
---[ RESPONSE ]--------------------------------------
HTTP/1.1 200 OK
Connection: close
Transfer-Encoding: chunked
Alt-Svc: h3-Q050=":443"; ma=2592000,h3-29=":443"; ma=2592000,h3-27=":443"; ma=2592000,h3-T051=":443"; ma=2592000,h3-T050=":443"; ma=2592000,h3-Q046=":443"; ma=2592000,h3-Q043=":443"; ma=2592000,quic=":443"; ma=2592000; v="46,43"
Cache-Control: private
Content-Type: application/json; charset=UTF-8
Date: Wed, 21 Oct 2020 10:25:23 GMT
Server: ESF
Vary: Origin
Vary: X-Origin
Vary: Referer
X-Content-Type-Options: nosniff
X-Frame-Options: SAMEORIGIN
X-Xss-Protection: 0

335
{
 "name": "projects/REDACTED/locations/us-central1/functions/tf-test-6onvdmxcmr",
 "description": "test function updated",
 "sourceArchiveUrl": "gs://tf-test-bucket-4020961597625764817/index_update.zip",
 "httpsTrigger": {
  "url": "https://us-central1-REDACTED.cloudfunctions.net/tf-test-6onvdmxcmr"
 },
 "status": "ACTIVE",
 "entryPoint": "helloGET",
 "timeout": "91s",
 "availableMemoryMb": 256,
 "serviceAccountEmail": "REDACTED@appspot.gserviceaccount.com",
 "updateTime": "2020-10-21T10:25:11.763Z",
 "versionId": "2",
 "labels": {
  "a-new-label": "a-new-label-value",
  "my-label": "my-updated-label-value"
 },
 "environmentVariables": {
  "NEW_ENV_VARIABLE": "new-env-variable-value",
  "TEST_ENV_VARIABLE": "test-env-variable-value"
 },
 "runtime": "nodejs10",
 "maxInstances": 15,
 "ingressSettings": "ALLOW_ALL",
 "buildId": "4d6ee76d-9bbd-41c3-ad44-5538986bacc7"
}
0


-----------------------------------------------------
2020/10/21 11:25:23 [DEBUG] Retry Transport: Stopping retries, last request failed with non-retryable error: empty non-retryable error received. This is a bug with the Terraform provider and should be reported as a GitHub issue in the provider repository.
2020/10/21 11:25:23 [DEBUG] Retry Transport: Returning after 1 attempts
2020/10/21 11:25:25 [INFO] Authenticating using DefaultClient...
2020/10/21 11:25:25 [INFO]   -- Scopes: [https://www.googleapis.com/auth/compute https://www.googleapis.com/auth/cloud-platform https://www.googleapis.com/auth/cloud-identity https://www.googleapis.com/auth/ndev.clouddns.readwrite https://www.googleapis.com/auth/devstorage.full_control https://www.googleapis.com/auth/userinfo.email]
2020/10/21 11:25:25 [INFO] Instantiating Google Cloud Billing client for path https://cloudbilling.googleapis.com/
2020/10/21 11:25:25 [INFO] Instantiating Google Cloud Build client for path https://cloudbuild.googleapis.com/
2020/10/21 11:25:25 [INFO] Instantiating Google Cloud BigQuery client for path https://bigquery.googleapis.com/bigquery/v2/
2020/10/21 11:25:25 [INFO] Instantiating Google Cloud CloudFunctions Client for path https://cloudfunctions.googleapis.com/
2020/10/21 11:25:25 [INFO] Instantiating Google Cloud BigtableAdmin for path https://bigtableadmin.googleapis.com/
2020/10/21 11:25:25 [INFO] Instantiating Google Cloud Source Repo client for path https://sourcerepo.googleapis.com/
2020/10/21 11:25:25 [INFO] Instantiating Google Cloud Spanner client for path https://spanner.googleapis.com/
2020/10/21 11:25:25 [INFO] Instantiating Google Cloud Dataproc client for path https://dataproc.googleapis.com/
2020/10/21 11:25:25 [INFO] Instantiating Google Cloud Dataproc Beta client for path https://dataproc.googleapis.com/
2020/10/21 11:25:25 [INFO] Instantiating Filestore client for path https://file.googleapis.com/
2020/10/21 11:25:25 [INFO] Instantiating Google Cloud IoT Core client for path 
2020/10/21 11:25:25 [INFO] Instantiating App Engine client for path https://appengine.googleapis.com/
2020/10/21 11:25:25 [INFO] Instantiating Cloud Composer client for path https://composer.googleapis.com/
2020/10/21 11:25:25 [INFO] Instantiating Service Networking client for path https://servicenetworking.googleapis.com/
2020/10/21 11:25:25 [INFO] Instantiating Google Cloud Storage Transfer client for path https://storagetransfer.googleapis.com/
2020/10/21 11:25:25 [INFO] Instantiating Google Cloud Healthcare client for path https://healthcare.googleapis.com/
2020/10/21 11:25:25 [INFO] Instantiating Google Cloud CloudIdentity client for path https://cloudidentity.googleapis.com/
    TestAccCloudFunctionsFunction_update: provider_test.go:267: Step 3/4 error: After applying this test step, the plan was not empty.
        stdout:
        
        
        An execution plan has been generated and is shown below.
        Resource actions are indicated with the following symbols:
          ~ update in-place
        
        Terraform will perform the following actions:
        
          # google_cloudfunctions_function.function will be updated in-place
          ~ resource "google_cloudfunctions_function" "function" {
                available_memory_mb         = 256
              ~ build_environment_variables = {
                  + "BUILD_NEW_ENV_VARIABLE"  = "new-env-variable-value"
                  + "BUILD_TEST_ENV_VARIABLE" = "test-env-variable-value"
                }
                description                 = "test function updated"
                entry_point                 = "helloGET"
                environment_variables       = {
                    "NEW_ENV_VARIABLE"  = "new-env-variable-value"
                    "TEST_ENV_VARIABLE" = "test-env-variable-value"
                }
                https_trigger_url           = "https://us-central1-REDACTED.cloudfunctions.net/tf-test-6onvdmxcmr"
                id                          = "projects/REDACTED/locations/us-central1/functions/tf-test-6onvdmxcmr"
                ingress_settings            = "ALLOW_ALL"
                labels                      = {
                    "a-new-label" = "a-new-label-value"
                    "my-label"    = "my-updated-label-value"
                }
                max_instances               = 15
                name                        = "tf-test-6onvdmxcmr"
                project                     = "REDACTED"
                region                      = "us-central1"
                runtime                     = "nodejs10"
                service_account_email       = "REDACTED@appspot.gserviceaccount.com"
                source_archive_bucket       = "tf-test-bucket-4020961597625764817"
                source_archive_object       = "index_update.zip"
                timeout                     = 91
                trigger_http                = true
            }
        
        Plan: 0 to add, 1 to change, 0 to destroy.

```

Comment (c2thorn) [reply third_party/terraform/tests/resource_cloudfunctions_function_test.go.erb:166]: Thanks! Makes sense as to why it only shows up in create/update methods.

**Review by c2thorn — APPROVED** [review]

**Pull request merged**